pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -41,6 +41,10 @@ pub struct InMemoryLayer {
     ///
     start_lsn: Lsn,
 
+    /// A buffer for serializing object during [`InMemoryLayer::put_value`].
+    /// This buffer is reused for each serialization to avoid additional malloc calls.
+    ser_buffer: Mutex<Vec<u8>>,
```

Comment:
This seems weird to me. Avoiding a malloc() is nice, but locking a Mutex isn't free either. Let's move this to InMemoryLayerInner, so that it doesn't need a separate Mutex to protect it. Or maybe even better, use `thread_local`. If it's thread-local, then we could potentially reuse the same scratch buffer in many different places.

Reply:
I used to store the buffer inside InMemoryLayerInner, but when calling `inner.file.write_blob(&buf)?`, it'll return error like: cannot borrow `buf` as immutable as `inner.file` is also borrowed as mutable.

`thread_local` looks good. Will try that out.